crates/node/src/network_server/handler/cluster_ctrl.rs

```diff
@@ -201,6 +202,37 @@ impl ClusterCtrlSvc for ClusterCtrlSvcHandler {
             })),
         }
     }
+
+    async fn seal_and_extend_chain(
+        &self,
+        request: Request<SealAndExtendChainRequest>,
+    ) -> Result<Response<()>, Status> {
+        let request = request.into_inner();
+        let Ok(kind) = ProtoProviderKind::try_from(request.provider) else {
+            return Err(Status::invalid_argument("Invalid provider kind"));
+        };
+
+        let kind: ProviderKind = kind
+            .as_str_name()
+            .parse()
+            .map_err(|_| Status::unimplemented("Provider type is not supported"))?;
+
+        self.controller_handle
+            .seal_and_extend_chain(
+                request.log_id.into(),
+                request.segment_index.map(SegmentIndex::from),
+                request
+                    .min_version
+                    .map(Version::from)
+                    .unwrap_or_else(|| Version::MIN),
+                kind,
+                request.params.into(),
+            )
+            .await
+            .map_err(|_| Status::aborted("Node is shutting down"))?
+            .map(|_| Response::new(()))
+            .map_err(|err| Status::internal(err.to_string()))
+    }
```

Comment:
I really wonder if this whole dance is necessary. In cluster_ctrl we already have a bifrost handle, why do we need to ask the controller service to do the operation for us?

Reply:
We do have a reference to `Bifrost` but not `BifrostAdmin`. The admin is owned by the controller service.

The BifrostAdmin seems to be cheap to create, but I don't believe I have all the bits in this context to create one to use it.

Reply:
The only thing you miss is MetadataWriter. That should be easy to plumb I guess. It's just a shame that we need to write so much code to do a simple thing.